app/tut/ga/nuair-a.md

```diff
@@ -0,0 +1,489 @@
+
+```{.setanta .numberLines}
+dathanna := ["glas", "buí", "bándearg", "gorm"]
+
+méideanna := [50, 100, 200]
+
+le i idir (0, fad@dathanna) {
+    dath := dathanna[i]
+
+    >-- Athraigh an dath
+    dath@stáitse(dath)
+
+    le j idir (0, fad@méideanna) {
+        méid := méideanna[j]
+    }
+}
+```
+
+Anois, sa lúb istigh tá na hathróige `dath` agus `méid` againn, agus is féidir linn ciorcail a
+tharraing, ach cén áit?
+
+Chun an ciorcail a tharraing i lár an stáitse is féidir linn `fad_x` agus `fad_y` a úsáid. Chun an
+[[lárphointe|midpoint]] a fháil níl le dhéanamh againn ach [[roinn|divide]] `fad_x` agus `fad_y`
+faoi 2.
+
+```{.setanta .numberLines}
+dathanna := ["glas", "buí", "bándearg", "gorm"]
+
+méideanna := [50, 100, 200]
+
+le i idir (0, fad@dathanna) {
+    dath := dathanna[i]
+
+    >-- Athraigh an dath
+    dath@stáitse(dath)
+
+    le j idir (0, fad@méideanna) {
+        méid := méideanna[j]
+
+        >-- Glan an stáitse.
+        glan@stáitse()
+
+        >-- Tarraing an ciorcal i lár an stáitse.
+        ciorcal@stáitse(fad_x@stáitse / 2, fad_y@stáitse / 2, méid)
+    }
+}
+```
+
+Faoi dheireadh, ba chóir dúinn [[fan ar feadh tamaill beag|wait for a short while]] tar éis dúinn an
+ciorcal a tharraing. Ba chóir dúinn é seo a dhéanamh mar mura ndéanaimid é beidh an ríomhchlár i
+bhfad ró thapa. Ní bheadh am againn chun na ciorcail ar leith a fheiceáil.
+
+Bain triail as!
+
+{{{s
+dathanna := ["glas", "buí", "bándearg", "gorm"]
+
+méideanna := [50, 100, 200]
+
+le i idir (0, fad@dathanna) {
+    dath := dathanna[i]
+
+    >-- Athraigh an dath
+    dath@stáitse(dath)
+
+    le j idir (0, fad@méideanna) {
+        méid := méideanna[j]
+
+        >-- Glan an stáitse.
+        glan@stáitse()
+
+        >-- Tarraing an ciorcal i lár an stáitse.
+        ciorcal@stáitse(fad_x@stáitse / 2, fad_y@stáitse / 2, méid)
+
+        >-- Codail ar feadh tamall beag.
+        codladh(400)
+    }
+}
+}}}
+
+# Stop! {#stad}
+
+Anois déanfaimid iarracht ríomhchlár nua a chruthú [[a ligeann don úsáideoir|that lets the user]]
+liosta téacs a chlóscríobh go dtí go gclóscríobhann siad "Stop", ansin scríobhfaidh an ríomhchlár an
+liosta ar ais ar an gconsól. Féach ar an GIF seo:
+
+![Ag clóscríobh liosta téacs](assets/ag-closcriobh-liosta-teacs.gif)
+
+Is féidir linn an gníomh `léigh`{.setanta} a úsáid chun [[ionchur an úsáideora|user input]] a fháil.
+Tosaímis le lúb `nuair-a fíor`{.setanta} agus feicimid níos déanaí
+[[conas éalú ón lúb|how to escape the loop]].
+
+```{.setanta .numberLines}
+>-- Cruthaigh liosta folamh.
+liosta := []
+
+nuair-a fíor {
+    >-- Faigh téacs ón úsáideoir.
+    téacs := léigh()
+
+    >-- Cuir an téacs leis an liosta.
+    liosta += [téacs]
+}
+```
+
+Má ritear an ríomhchlár sin, leanfaidh sé ar aghaidh go deo ag léigh téacs ón gconsól. Ba mhaith
+linn scoir ón lúb nuair a scríobhann an úsáideoir "Stop". Conas is féidir linn é sin a dhéanamh?
+
+## DIY
```

Comment:
Cad is brí le DIY anseo?

Reply:
An rud céanna a chiallaíonn sé as Béarla (Do It Yourself). Choimeád mé é as Béarla mar is saghas ainmfhocal dílis é. An bhfuil moladh eile agat?

Reply:
Níl. Cheap mé go raibh sé i gceist agat rud éigin níos iomláine a scríobh. "Déan féin é" b'fhéidir